### PR TITLE
ROX-12624: GetCollection REST endpoint and service framework

### PR DESCRIPTION
**Pull request by keyallis**
## Description

Add the service layer framework for collections and implement the GetCollection endpoint.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added

## Testing Performed

Verified that the API docs include the service and the implemented endpoint. Queried the endpoint using roxcurl.

### CODE REVIEW

File: central/resourcecollection/service/service.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+type Service interface {
+	grpc.APIService
+
+	AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error)
+
+	v1.CollectionServiceServer
+}
+
+// New returns a new Service instance using the given DataStore.
+func New(
```
Comment (theencee) [inline central/resourcecollection/service/service.go:21]: [nit] Is this an artifact of gofmt? If not, IMO it looks nicer on one line

**Review by theencee — APPROVED** [review]
lgtm. For the test failures try retest?

File: central/resourcecollection/service/service.go (hunk 1 above)
Comment (keyallis) [reply central/resourcecollection/service/service.go:21]: This was a copy/paste from another implementation that had multiple fields and the formatting never got changed after trimming down args. I'm going to see if the tests pass before I push this small change

Comment (theencee) [reply central/resourcecollection/service/service.go:21]: Not a big deal. You can combine this with another PR when you implement other endpoints.

**Comment by keyallis** [comment]
Fixed the nit and did a rebase on master to see if that fixes the CI issues

**Pull request merged**